submissions/VolodymyrRutskyi/tiny-js-world/index.js

```diff
@@ -1,90 +1,52 @@
-/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
-   Complete the below for code reviewers' convenience:
+class Inhabitant {
+   constructor(species, name, gender, legs, saying){
+      this.species = species;
+      this.name = name;
+      this.gender = gender;
+      this.legs = legs;
+      this.saying = saying;
+      this.props = ['species', 'name', 'gender', 'legs', 'saying']
+   }
+   showProps() {
+      return this.props.map(prop => this[prop]).join('; ');
+   }
+}
+
+class Human extends Inhabitant{
+   constructor (name, gender, saying){
+      super('human', name, gender, 2, saying);
+      this.hands = 2;
+      this.props = [...this.props, 'hands']
```

Comment:
It is generally a bad practice to directly change properties that are owned by the base class.
Every class should handle its own properties i.e. encapsulate both properties and behaviour.
The same refers to `Inhabitant#showProps`.
Let every class handle properties it owns.
Use method overloading to provide seamless user experience.
You will benefit from explicit call to inherited version of the methos via `super`.
Task and original project materials offer links to the articles that you will find helpful.

Reply:
I have fixed